### ISSUE TITLE
"fanout.fpp.push" instead of "xmpp.buddycloud.push.item"

### ISSUE THREAD

**Issue by robotnic** [xmpp-ftw/xmpp-ftw-buddycloud#32]
I'm publishing to a buddycloud node

```
{"type":0,"data":["xmpp.buddycloud.publish",{"node":"/user/maria@laos.buddycloud.com/posts","content":{"atom":{"content":"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"}}}],"id":7}
```

and get the following notification

```

:{"type":0,"data":["fanout.fpp.push",{"id":"tag:null@cs.buddycloud.com,/user/maria@laos.buddycloud.com/posts,38fc616d-bb6c-478e-a0a9-f568d847bd5a","entry":{"atom":{"title":"Post","id":"tag:null@cs.buddycloud.com,/user/maria@laos.buddycloud.com/posts,38fc616d-bb6c-478e-a0a9-f568d847bd5a","updated":"2015-01-19T22:28:20.938Z","published":"2015-01-19T22:28:20.938Z","content":{"content":"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa","type":"text"},"author":{"name":"maria@laos.buddycloud.com","uri":"acct:maria@laos.buddycloud.com"}},"activity":{"verb":"post","object":{"object-type":"note"},"author":{"object-type":"person"}}},"realm":"maria@laos.buddycloud.com","channel":"user"}]}
```

I'm expecting "xmpp.buddycloud.push.item"

https://xmpp-ftw.jit.su/manual/extensions/buddycloud/#notification-of-new-messages

The XML stanza

```

<id>tag:null@cs.buddycloud.com,/user/maria@laos.buddycloud.com/posts,3a81c5c3-ef0b-4e27-b0bd-d564dd50535c</id><published>2015-01-19T22:35:58.998Z</published><updated>2015-01-19T22:35:58.998Z</updated><author><name>maria@laos.buddycloud.com</name><uri>acct:maria@laos.buddycloud.com</uri><object-type xmlns="http://activitystrea.ms/spec/1.0/">person</object-type></author><title>Post</title><content type="text">aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa</content><verb xmlns="http://activitystrea.ms/spec/1.0/">post</verb><object xmlns="http://activitystrea.ms/spec/1.0/"><object-type xmlns="http://activitystrea.ms/spec/1.0/">note</object-type></object></entry></item></items></event></message>
```

**Comment by robotnic** [comment]
If I use xmpp-ftw at http//loas.buddycloud.com I get this:

```
3:::{"type":0,"data":["xmpp.buddycloud.push.item",{"node":"/user/seppl@laos.buddycloud.com/posts","id":"tag:null@cs.buddycloud.com,/user/seppl@laos.buddycloud.com/posts,6ad8217d-e7d1-4d2d-8b3d-982ee67181ba","entry":{"atom":{"title":"Post","id":"tag:null@cs.buddycloud.com,/user/seppl@laos.buddycloud.com/posts,6ad8217d-e7d1-4d2d-8b3d-982ee67181ba","updated":"2015-01-19T23:11:07.944Z","published":"2015-01-19T23:11:07.944Z","content":{"content":"oha","type":"text"},"author":{"name":"seppl@laos.buddycloud.com","uri":"acct:seppl@laos.buddycloud.com"}},"activity":{"verb":"post","object":{"object-type":"comment"},"author":{"object-type":"person"}},"in-reply-to":{"ref":"591ed835-e588-4ff1-9566-dbfd05a84a7f"}}}]}
```

**Comment by lloydwatkin** [comment]
Yes, that's to be expected as the demo application also includes the fanout plugin which 'wins' the race to pick up the incoming stanza.

If running locally take out this line from xmpp-ftw-demo https://github.com/xmpp-ftw/xmpp-ftw-demo/blob/master/index.js#L82

**Comment by robotnic** [comment]
Now the problem is

If fanout is active
"xmpp.pubsub.push.affiliation"

If fanout is not active
"xmpp.buddycloud.push.affiliation"

**Comment by lloydwatkin** [comment]
I've just published a new version of `xmpp-ftw-fanout@0.1.8`, that should prevent your affiliation push issues (although it'll still intercept publication pushes).

**Comment by robotnic** [comment]
I have a working version for localhost.  But Simon is not able install that. xmpp-ftw is down since one month.
